src/Datadog.AutoInstrumentation.NativeLoader/dynamic_dispatcher.cpp

```diff
@@ -0,0 +1,332 @@
+#elif LINUX
+const std::string currentOsArch = "linux-x86";
+#elif MACOS
+const std::string currentOsArch = "osx-x86";
+#else
+#error "currentOsArch not defined."
+#endif
+
+#elif ARM64
+
+#if _WINDOWS
+const std::string currentOsArch = "win-arm64";
+#elif LINUX
+const std::string currentOsArch = "linux-arm64";
+#elif MACOS
+const std::string currentOsArch = "osx-arm64";
+#else
+#error "currentOsArch not defined."
+#endif
+
+#elif ARM
+
+#if _WINDOWS
+const std::string currentOsArch = "win-arm";
+#elif LINUX
+const std::string currentOsArch = "linux-arm";
+#elif MACOS
+const std::string currentOsArch = "osx-arm";
+#else
+#error "currentOsArch not defined."
+#endif
+
+#else
+#error "currentOsArch not defined."
+#endif
+
+namespace datadog::shared::nativeloader
+{
+
+    // ************************************************************************
+
+    //
+    // public
+    //
+
+    DynamicDispatcherImpl::DynamicDispatcherImpl()
+    {
+        m_continuousProfilerInstance = nullptr;
+        m_tracerInstance = nullptr;
+        m_customInstance = nullptr;
+    }
+
+    void DynamicDispatcherImpl::LoadConfiguration(std::string configFilePath)
+    {
+        if (!std::filesystem::exists(configFilePath))
+        {
+            Warn("Configuration file doesn't exist.");
+            return;
+        }
+
+        std::ifstream t(configFilePath);
+
+        // Gets the configuration file folder
+        std::filesystem::path configFolder = std::filesystem::path(configFilePath).remove_filename();
+        Debug("Config Folder: ", configFolder);
+
+        // Get the current path
+        std::filesystem::path oldCurrentPath = std::filesystem::current_path();
+        Debug("Current Path: ", oldCurrentPath);
+
+        // Set the current path to the configuration folder (to allow relative paths)
+        std::filesystem::current_path(configFolder);
+
+        const std::string allOsArch[12] = {
+            "win-x64", "linux-x64", "osx-x64",
+            "win-x86", "linux-x86", "osx-x86",
+            "win-arm64", "linux-arm64", "osx-arm64",
+            "win-arm", "linux-arm", "osx-arm",
+        };
+
+        while (t)
+        {
+            std::string line;
+            std::getline(t, line);
+            line = Trim(line);
+            if (line.length() != 0)
+            {
+                Debug(line);
+
+                if (line[0] == '#')
+                {
+                    continue;
+                }
+
+                std::vector<std::string> lineArray = Split(line, ';');
+                if (lineArray.size() != 4)
+                {
+                    Warn("Invalid line: ", line);
+                }
+
+                std::string type = lineArray[0];
+                std::string idValue = lineArray[1];
+                std::string osArchValue = lineArray[2];
+                std::string filepathValue = lineArray[3];
+
+                if (type == "TRACER" || type == "CP" || type == "CUSTOM")
```

Comment:
Can CP be PROFILER. its easier for a customer to understand.
or ContinuousProfiler or similar..?

Reply:
done in 95e4147